File: app/src/main/java/com/hawksjamesf/simpleweather/util/KeyboardUtil.java
package com.hawksjamesf.simpleweather.util;

import android.app.Activity;
import android.content.Context;
import android.view.View;
import android.view.inputmethod.InputMethodManager;

import com.hawksjamesf.simpleweather.SimpleWeatherApplication;

/**
 * Copyright ® $ 2017
 * All right reserved.
 *
 * @author: hawks.jamesf
 * @since: Oct/24/2018  Wed
 */
public class KeyboardUtil {
    public static void hideSoftInput(Activity activity) {
        InputMethodManager manager = (InputMethodManager) SimpleWeatherApplication.getInstance().getSystemService(Context.INPUT_METHOD_SERVICE);
        View currentView = activity.getCurrentFocus();
        if (currentView == null) currentView = new View(activity);
        manager.hideSoftInputFromWindow(currentView.getWindowToken(), 0);
    }
}

File: app/src/main/kotlin/com/hawksjamesf.simpleweather/util/KeyboardUtil.kt
package com.hawksjamesf.simpleweather.util

import android.app.Activity
import android.content.Context
import android.view.inputmethod.InputMethodManager

/**
 * Copyright ® $ 2017
 * All right reserved.
 *
 * @author: hawks.jamesf
 * @since: Oct/24/2018  Wed
 */
fun Activity.hideSoftInput() {
    currentFocus?.windowToken?.let {
        val manager = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
        manager.hideSoftInputFromWindow(it, 0)
    }
}